### PR TITLE
associate host with facet

### PR DESCRIPTION
**Pull request by timogoebel**

### CODE REVIEW

File: app/models/concerns/foreman_build_history/host_extensions.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@ module HostExtensions
       after_build :set_build_start
       before_provision :set_build_finish
       before_save :build_history_handle_host_creation
+      has_one :build_history_facet, :foreign_key => 'host_id'
```
Comment (ShimShtein) [inline app/models/concerns/foreman_build_history/host_extensions.rb:10]: First, thanks for looking into this plugin, I really appreciate it!

This change is unnecessary though, facet framework handles facet relations [automatically](https://github.com/theforeman/foreman/blob/develop/app/models/concerns/facets/managed_host_extensions.rb#L38)

Comment (timogoebel) [reply app/models/concerns/foreman_build_history/host_extensions.rb:10]: I really like the idea of the plugin. We should make it official at some point.
It broke without the change in my local dev environment, though. Is there something else wrong?

```
NameError (undefined local variable or method `build_history_facet' for #<Host::Managed:0x007fedcdb3eda0>
```

Comment (ShimShtein) [reply app/models/concerns/foreman_build_history/host_extensions.rb:10]: My first hunch would be reloading in development environment.
Take a look at `Facets.registered_facets`. If you don't see it there, it's probably reloading.

Comment (timogoebel) [reply app/models/concerns/foreman_build_history/host_extensions.rb:10]: Yeah, that probably was it. We should make facts more reliable to reloading, then.

Comment (ShimShtein) [reply app/models/concerns/foreman_build_history/host_extensions.rb:10]: Sure!
Open a ticket, I'll fix it.

Comment (timogoebel) [reply app/models/concerns/foreman_build_history/host_extensions.rb:10]: @ShimShtein: Thanks, http://projects.theforeman.org/issues/17309

**Pull request closed without merging**